File: python/7 kyu/257.py
# https://www.codewars.com/kata/5982619d2671576e90000017/train/python

# Passed

def sponge_meme(s):
    result = str()
    transform_to_upper = True
    for char in s:
        result += char.upper() if transform_to_upper else char.lower()
        transform_to_upper = not transform_to_upper
        
    return result

output = sponge_meme("stop Making spongebob Memes!")
print(output)

File: nodejs/7 kyu/257.js
// https://www.codewars.com/kata/5b049d57de4c7f6a6c0001d7/train/javascript

// Passed

function apparently(string) {
  let words = string.split(" ");
  let apparentlyWords = [];

  for (let idx = 0; idx < words.length; idx++) {
    apparentlyWords.push(words[idx]);
    if (
      ["and", "but"].includes(words[idx].toLowerCase()) &&
      (idx === words.length - 1 ||
        words[idx + 1].toLowerCase() !== "apparently")
    ) {
      apparentlyWords.push("apparently");
    }
  }

  let apparentlySentence = apparentlyWords.join(" ");
  return apparentlySentence;
}

const output = apparently(
  "It was great and I have never been on live television before but sometimes I dont watch this.",
);
console.log(output);